### PR TITLE
Closes #1499 - Update Requirements for Pandas

### LINKED ISSUE #1499: Update Requirements for Pandas
Update pandas minimum to 1.4.0.

### PR DESCRIPTION
**Pull request by Ethan-DeBandi99**
Closes #1499

- Updates `setup.py`, `arkouda-env-dev.yml`, `arkouda-env.yml`, and `pydoc/requirements.txt` to include updated dependencies for Pandas. We updated to require `pandas>=1.4.0` which required other minimum versions be updated, namely `numpy>=1.18.5`.
- Adds `REQUIREMENTS.md` file for easy identification of requirements and information on how to install dependencies and update them when they are out of date.

### CODE REVIEW

**Review by mhmerrill — APPROVED** [review]
looks ok to me

**Review by jeichert60 — COMMENTED** [review]
Just a couple of typos:

- Dependencies can be install using `Anaconda` (Recommended) or `pip`. 
+ Dependencies can be installed using `Anaconda` (Recommended) or `pip`. 

- Users Enviorment YAML: `arkouda-env.yml`
- Developer Enviornment YAML: `arkouda-env-dev.yml`
Change enviornment to environment

**Review by jeichert60 — APPROVED** [review]

File: REQUIREMENTS.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+- `Sphinx`
+- `sphinx-argparse`
+- `sphinx-autoapi`
+- `typed-ast`
+- `mypy>=0.931`
+- `flake8`
+
+# Installing Dependencies
+
+Dependencies can be install using `Anaconda` (Recommended) or `pip`.
```
Comment (pierce314159) [inline REQUIREMENTS.md:30]: typo `install` to `installed`

File: REQUIREMENTS.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+# Installing Dependencies
+
+Dependencies can be install using `Anaconda` (Recommended) or `pip`. 
+
+## Using Anaconda
+Arkouda provides 2 files for installing dependencies, one for users and one for developers. 
+
+- Users Enviorment YAML: `arkouda-env.yml`
+- Developer Enviornment YAML: `arkouda-env-dev.yml`
```
Comment (pierce314159) [inline REQUIREMENTS.md:36]: typo `Enviorment` to `Environment`

**Review by pierce314159 — APPROVED** [review]
just highlighting the lines from jim's review. It all looks good to me

File: REQUIREMENTS.md (hunk 1 above)
Comment (Ethan-DeBandi99) [reply REQUIREMENTS.md]: done

File: REQUIREMENTS.md (hunk 2 above)
Comment (Ethan-DeBandi99) [reply REQUIREMENTS.md]: done

**Pull request merged**